sdk/resourcemanager/Azure.ResourceManager/src/ArmResource.cs

```diff
@@ -42,6 +42,16 @@ protected internal ArmResource(ArmClient client, ResourceIdentifier id)
             Client = client;
             Id = id;
         }
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ArmResource"/> class.
+        /// </summary>
+        /// <param name="client"> The <see cref="ArmClient"/> this resource client should be created from. </param>
+        protected internal ArmResource(ArmClient client)
+        {
+            Argument.AssertNotNull(client, nameof(client));
+
+            Client = client;
+        }
```

Comment:
Add this constructor as a quick workaround that `IOperationSource` of a resource class needs to use the `ArmClient`.